pexpect/pty_spawn.py

```diff
@@ -430,11 +452,7 @@ def read_nonblocking(self, size=1, timeout=-1):
         # For this case, I test isalive() before doing any reading.
         # If isalive() is false, then I pretend that this is the same as EOF.
```

Comment:
> This comment appears to be saying that on Solaris, read on a pty after EOF will block

The question is really: what will `select()` do in this situation? If `select()` simply times out, we are safe.